library/src/main/java/com/mux/video/upload/internal/UploadMetrics.kt

```diff
@@ -13,14 +13,153 @@ import okhttp3.Interceptor
 import okhttp3.MediaType.Companion.toMediaType
 import okhttp3.Request
 import okhttp3.RequestBody.Companion.toRequestBody
+import org.json.JSONArray
 import org.json.JSONObject
+import java.text.SimpleDateFormat
 import java.util.*
 import java.util.concurrent.TimeUnit
 
+
 internal class UploadMetrics private constructor() {
 
   private val logger get() = MuxUploadSdk.logger
 
+
+  private fun formatMilliseconds(ms:Long):String {
+    return String.format("%02d:%02d:%02d",
+      TimeUnit.MILLISECONDS.toHours(ms),
+      TimeUnit.MILLISECONDS.toMinutes(ms) -
+              TimeUnit.HOURS.toMinutes(TimeUnit.MILLISECONDS.toHours(ms)), // The change is in this line
+      TimeUnit.MILLISECONDS.toSeconds(ms) -
+              TimeUnit.MINUTES.toSeconds(TimeUnit.MILLISECONDS.toMinutes(ms)));
+  }
+
+  private suspend fun getEventInfo(startTimeMillis: Long,
+                           endTimeMillis: Long,
+                           inputFileDurationMs: Long,
+                           uploadInfo: UploadInfo): JSONObject {
+    val iso8601Sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ", Locale.getDefault())
+    var videoDuration = inputFileDurationMs
+    if (inputFileDurationMs <= 0) {
+        videoDuration = withContext(Dispatchers.IO) {
+        try {
+          MediaMetadataRetriever().use { retriever ->
+            retriever.setDataSource(uploadInfo.inputFile.absolutePath)
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)?.toInt()
+          }
+        } catch (e: Exception) {
+          MuxUploadSdk.logger.e("UploadMetrics", "Failed to get video duration", e)
+          null
+        }
+      }!!.toLong()
+    }
+    return JSONObject().apply {
+      put("standardization_start_time", iso8601Sdf.format(startTimeMillis)) // ISO8601
+      put("standardization_end_time", iso8601Sdf.format(endTimeMillis)) // ISO8601
+      put("input_size", uploadInfo.inputFile.length())
+      put("input_duration", formatMilliseconds(videoDuration)) // HH:mm:ss
+      put("upload_url", uploadInfo.remoteUri.toString())
+      put("sdk_version", BuildConfig.LIB_VERSION)
+      put("os_name", "Android")
+      put("os_version", Build.VERSION.RELEASE)
+      put("device_model", Build.MODEL)
+      put("app_name", appName)
+      put("app_version", appVersion)
+      put("region_code", Locale.getDefault().country)
+    }
+  }
+
+  private suspend fun sendPost(body:JSONObject) {
+    val httpClient = MuxUploadSdk.httpClient()
+    val request = Request.Builder()
+      .url("https://mobile.muxanalytics.com")
+      .method("POST", body.toString().toRequestBody("application/json".toMediaType()))
+      .build()
+    // The HTTP Client will log if this fails or succeeds
+    withContext(Dispatchers.IO) { httpClient.newCall(request).execute() }
```

Comment:
it looks like the answer to my general question is no from this line, but just want confirmation